File: attence_system_backend_java/attence_system_backend/src/main/java/attence_system_backend/attence_system_backend/models/Employee.java
package attence_system_backend.attence_system_backend.models;

import java.util.List;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "employees")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // --- Basic Information ---
    private String fullName;
    
    @Column(unique = true, nullable = false)
    private String employeeId;
    
    private String department;
    private String position;
    private String joinDate; 
    private String workStatus;
    private String profileImageUrl; 

    // --- Authentication & Account (Formerly in User model) ---
    @Column(unique = true, nullable = false)
    private String email;

    @Column(nullable = false)
    private String password;

    private String role; // ADMIN, EMPLOYEE

    // -------------------------------
    // NEW PROFILE INFORMATION (Mobile App)
    // -------------------------------

    // Personal
    private String khmerName;
    private String gender;
    private String birthDate;
    private String relationshipStatus;

    // Contact
    private String phoneNumber;
    private String idCard;

    // Work extra
    private String residency;
    private String employmentType;

     // Relationship with attendance
    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL)
    private List<Attendance> attendances;

}

File: attence_system_Mobile_app_flutter/lib/page/employee.dart
class Employee {
  final String fullName;
  final String employeeId;
  // final String? position;
  final String? profileImageUrl;

  Employee({
    required this.fullName,
    required this.employeeId,
    // this.position,
    this.profileImageUrl,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      fullName: json['fullName'] ?? '',
      employeeId: json['employeeId'] ?? '',
      // position: json['position'],
      profileImageUrl: json['profileImageUrl'],
    );
  }
}